app/src/main/java/org/dslul/openboard/inputmethod/keyboard/KeyboardView.java

```diff
@@ -517,23 +534,37 @@ protected void onDrawKeyTopVisuals(@NonNull final Key key, @NonNull final Canvas
         }
     }
 
-    // Draw popup hint "..." at the bottom right corner of the key.
     protected void drawKeyPopupHint(@NonNull final Key key, @NonNull final Canvas canvas,
             @NonNull final Paint paint, @NonNull final KeyDrawParams params) {
         if (TextUtils.isEmpty(mKeyPopupHintLetter)) {
             return;
         }
+        final SharedPreferences prefs = DeviceProtectedUtils.getSharedPreferences(getContext());
+        final String style = prefs.getString(Settings.PREF_THEME_STYLE, STYLE_MATERIAL);
         final int keyWidth = key.getDrawWidth();
         final int keyHeight = key.getHeight();
-
+        final float labelCharWidth = TypefaceUtils.getReferenceCharWidth(paint);
+        final float hintBaseline = paint.ascent();
         paint.setTypeface(params.mTypeface);
         paint.setTextSize(params.mHintLetterSize);
         paint.setColor(params.mHintLabelColor);
         paint.setTextAlign(Align.CENTER);
-        final float hintX = keyWidth - mKeyHintLetterPadding
-                - TypefaceUtils.getReferenceCharWidth(paint) / 2.0f;
+        // For Rounded style: draw popup hint "..." in the center of the functional key.
+        final float hintX1 = keyWidth / 2.0f;
+        // For Rounded style: draw popup hint "..." in the bottom right corner of the space bar.
+        final float hintX2 = keyWidth + hintBaseline + labelCharWidth * 0.1f;
+        // For other styles: draw popup hint "..." in the bottom right corner of the key.
+        final float hintX3 = keyWidth - mKeyHintLetterPadding - labelCharWidth / 2.0f;
         final float hintY = keyHeight - mKeyPopupHintLetterPadding;
-        canvas.drawText(mKeyPopupHintLetter, hintX, hintY, paint);
+        if (style.equals(STYLE_ROUNDED)) {
+            if (key.isFunctional() || key.isActionKey()) {
+                canvas.drawText(mKeyPopupHintLetter, hintX1, hintY, paint); // For functional keys
+            } else {
+                canvas.drawText(mKeyPopupHintLetter, hintX2, hintY, paint); // For space bar
+            }
+        } else {
+            canvas.drawText(mKeyPopupHintLetter, hintX3, hintY, paint); // For other styles
+        }
```

Comment:
It's probably irrelevant for performance, but there is no need to calculate `hintX` for every case when only one is used. (similar for `adjustmentY1` and `adjustmentY2` above)